libbeat/processors/processor.go

```diff
@@ -0,0 +1,88 @@
+package processors
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type Processors struct {
+	list []ProcessorInterface
+}
+
+func New(config ProcessorPluginConfig) (*Processors, error) {
+
+	processors := Processors{}
+
+	for _, processor := range config {
+
+		if len(processor) != 1 {
+			return nil, fmt.Errorf("each processor needs to have exactly one action, but found %d actions.",
+				len(processor))
+		}
+
+		for processorName, cfg := range processor {
+
+			constructor, exists := constructors[processorName]
+			if !exists {
+				return nil, fmt.Errorf("the processor %s doesn't exist", processorName)
+			}
+
+			plugin, err := constructor(cfg)
+			if err != nil {
+				return nil, err
+			}
+
+			processors.addProcessor(plugin)
+		}
+	}
+
+	logp.Debug("processors", "Processors: %v", processors)
+	return &processors, nil
+}
+
+func (processors *Processors) addProcessor(p ProcessorInterface) {
+
+	if processors.list == nil {
+		processors.list = []ProcessorInterface{}
+	}
```

Comment:
this if-block is not required. appending to `nil` ensures array is allocated first time.